### ISSUE TITLE
VS Code instance freezes after terminating vscode debugging session on javascript HttpTrigger function

### ISSUE THREAD

**Issue by SDP190** [microsoft/vscode-azurefunctions#1401]
I have latest vscode version running on windows 10.
I am able to run Azure Functions locally on VS code using AzureFunctions ext. for VS Code, however when I run the debugger (F5) on my JavaScript Azure Function V2 (HttpTrigger), then when I stop it, very often it causes VS Code instance to crash and freeze.

Any help on this?

Its a standard project created with VS Code.

**VS Code Info (Copied from About window):**
Version: 1.36.1 (system setup)
Commit: 2213894ea0415ee8c85c5eea0d0ff81ecc191529
Date: 2019-07-08T22:59:35.033Z
Electron: 4.2.5
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.18362

**Node.js Version:**
`node -v  --> v10.15.3`

**Comment by EricJizbaMSFT** [comment]
@SDP190 What language is your project? Does it show any errors?

**Comment by SDP190** [comment]
My language is JavaScript node.js
It does not show any errors!

**Comment by EricJizbaMSFT** [comment]
Is your project a standard project created from VS Code? If not - is it open source and can you share a link?

Also just to confirm - can you copy your exact VS Code version and your Node.js version

**Comment by SDP190** [comment]
@EricJizbaMSFT See original comment updated

**Comment by askpatrickw** [comment]
I experience this intermittently as well with Python.

standard project created with vscode

> VS Code Version: 1.36.1 (user setup)
> Commit: 2213894ea0415ee8c85c5eea0d0ff81ecc191529
> Date: 2019-07-08T22:59:35.033Z
> Electron: 4.2.5
> Chrome: 69.0.3497.128
> Node.js: 10.11.0
> V8: 6.9.427.31-electron.0
> OS: Windows_NT x64 10.0.18362

> (.env) PS: github\project> python --version
> Python 3.6.8

**Comment by EricJizbaMSFT** [comment]
@SDP190 @askpatrickw If you can get concrete repro steps that would be great. I'm having a hard time repro-ing. That being said - I actually submitted a PR a little bit ago that might affect this experience: https://github.com/microsoft/vscode-azurefunctions/pull/1389 (The PR title says C#, but it should affect other languages as well)

If you have the time, please try out the latest build from master with these instructions and let me know if it helps: https://aka.ms/azCodeInsiders

**Comment by SDP190** [comment]
@EricJizbaMSFT, I cant tell you exactly what the problem is because it does not occur always I cannot pinpoint any cause, however what I can tell you is to play around with a HttpTrigger javascript function calling the the function with http calls multiple times (1 after another), also maybe try putting breakpoints in the code, then try 'Disconnect' the debugging session at verius times during either when its in break mode either while it is running.

I will maybe try later minimizing my code to see if it maybe caused by my own code, (I use mssql npm package for sql connections, and also have multiple awaits, however I dont believe its related to this).

**Comment by SDP190** [comment]
I tried the latest build, did not solve the problem.

**Comment by askpatrickw** [comment]
I have a very simple python function with an http binding for GET only. It returns a file from blob storage. The file is retrieved using the requests library and the url is put together with a url parmeter value placed in an f-string.

requests is the only thing in my requirements.txt

I can run my project, make a request. Stop the debugger and start it again and it will hang.
If I kill the terminal window it seems to prevent that.

**Comment by SDP190** [comment]
@askpatrickw, my HttpTrigger is configured for 'POST' only.

**Comment by fiveisprime** [comment]
I saw this while working with @sbatten. Seems to repro 100% for him if you want to get in touch with each other.

**Comment by EricJizbaMSFT** [comment]
@SDP190 @askpatrickw @sbatten FYI we just released v0.19.0 of the extension. It has a new setting `azureFunctions.stopFuncTaskPostDebug` which you can set to `false` if you don't want us to stop the functions host after you stop debugging (which may be causing the crash).

Let us know if that helps. Also let us know if you still see a crash if you manually stop the functions host (either through ctrl-c or the trash can icon).

**Comment by EricJizbaMSFT** [comment]
Awesome - are you manually stopping the func host task and does that work without crashing? Also what version of the func cli are you on?

**Comment by sevenam** [comment]
Seen the same problem with python on az cli v2.0.73 and multiple versions of vs code.
This seem to have fixed it for me too (not happened since I disabled this):

> azureFunctions.stopFuncTaskPostDebug = false

Manually stopping it with ctrl-c has not caused any freezing yet either.

**Comment by jsacapdev** [comment]
This is happening for me consistently. My project is a Function developed on C#. A full sample project can be found here:

https://github.com/jsacapdev/steel.function

While adding `"azureFunctions.stopFuncTaskPostDebug": false` to `settings.json` this is not ideal as then the assemblies become locked on subsequent builds. 

```
Build started 02/06/2020 07:31:55.
     1>Project "C:\git\steel.function\st.cfg.str.rbac.sln" on node 1 (Clean target(s)).
     1>ValidateSolutionConfiguration:
         Building solution configuration "Debug|Any CPU".
     1>Project "C:\git\steel.function\st.cfg.str.rbac.sln" (1) is building "C:\git\steel.function\src\ConfigFunctionApi\ConfigFunctionApi.csproj" (2) on node 1 (Clean target(s)).
     2>CoreClean:
         Deleting file "C:\git\steel.function\src\ConfigFunctionApi\bin\Debug\netcoreapp3.1\host.json".
         Deleting file "C:\git\steel.function\src\ConfigFunctionApi\bin\Debug\netcoreapp3.1\local.settings.json".
         Deleting file "C:\git\steel.function\src\ConfigFunctionApi\bin\Debug\netcoreapp3.1\ConfigFunctionApi.deps.json".
         Deleting file "C:\git\steel.function\src\ConfigFunctionApi\bin\Debug\netcoreapp3.1\bin\ConfigFunctionApi.dll".
     2>C:\Program Files\dotnet\sdk\3.1.300\Microsoft.Common.CurrentVersion.targets(5148,5): warning MSB3061: Unable to delete file "C:\git\steel.function\src\ConfigFunctionApi\bin\Debug\netcoreapp3.1\bin\ConfigFunctionApi.dll". Access to the path 'C:\git\steel.function\src\ConfigFunctionApi\bin\Debug\netcoreapp3.1\bin\ConfigFunctionApi.dll' is denied. [C:\git\steel.function\src\ConfigFunctionApi\ConfigFunctionApi.csproj]
         Deleting file "C:\git\steel.function\src\ConfigFunctionApi\bin\Debug\netcoreapp3.1\bin\Microsoft.ApplicationInsights.dll".
         Deleting file "C:\git\steel.function\src\ConfigFunctionApi\bin\Debug\netcoreapp3.1\bin\Microsoft.Azure.Functions.Extensions.dll".
```

This development experience is less than ideal.

I am getting the same experience on insiders.

Please let me know if there is anything else, I can provide to help diagnose this issue. My machine is a Surface Book 2 running Windows 10.

**Comment by EricJizbaMSFT** [comment]
@jsacapdev what version of the func cli do you have? Are you on Windows 10 Home or Pro?

Also can you go to the "About" thing for VS Code and copy all the versions from there

**Comment by jsacapdev** [comment]
@EricJizbaMSFT i believe i have the following version of the cli (let me know if this is not correct):

```
PS C:\> func --version
3.0.2534
```

I am on `Windows 10 Pro 1909 (Version 10.0.18363 Build 18363)`.

VS Code:
```
Version: 1.45.1 (user setup)
Commit: 5763d909d5f12fe19f215cbfdd29a91c0fa9208a
Date: 2020-05-14T08:27:35.169Z
Electron: 7.2.4
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363
```

Insiders:
```
Version: 1.46.0-insider (user setup)
Commit: 2c1871d538c64df71af4902deba47d7fa4b53f27
Date: 2020-06-01T10:34:11.096Z
Electron: 7.3.0
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363
```

**Comment by cloudappsguy** [comment]
Hi @EricJizbaMSFT - I have the same problem as @jsacapdev:

Running an Azure Function App (C#) locally, in debug mode.  Set some break points.  Call the local FA via Postman a time or three.

Then, attempting to stop debugging via disconnecting, the UI freezes and eventually (10-20 seconds later) VSCode pops a window stating the environment is non responsive and asks if I want to restart it.

**Function App version:**
3.0.2630

**VS Code:**

Version: 1.47.3 (user setup)
Commit: 91899dcef7b8110878ea59626991a18c8a6a1b3e
Date: 2020-07-23T13:12:49.994Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363

Happy to provide any other details needed.

**Comment by bijeebuss** [comment]
any updates on this. I having the same issue with a C# function app with a timer trigger

**Comment by pacorreia** [comment]
same thing here, this as started with latest azure function core tools

**Comment by bijeebuss** [comment]
mine doesn't freeze as long as I kill it with ctrl-c in the terminal window

**Comment by stefanushinardi** [comment]
@EricJizbaMSFT, is there any update to this investigation? thanks

**Comment by erharvey** [comment]
Exact same issue on me side with the latest core tools. CTRL-C  in the terminal is the only way to not freeze vcode.

**Comment by XsongyangX** [comment]
I'm experiencing this issue from time to time on Python debugger as well.

**Comment by prkhandelwal** [comment]
Experiencing the same issue with Python debugger

**Comment by pmillot** [comment]
Exact same issue here every time, with the latest versions of everything.

**Comment by mpriem** [comment]
Same issue here. latest builds of tools and client. Doesn't matter what language or architecture. Hangs on x32/x64 with JS/C#/PS. It happens less if you close the terminal once the function ends.

**Comment by ejizba** [comment]
Okay folks just released v1.1.0 of the extension which should hopefully fix this issue. I'm going to go ahead and close it, but let me know if you're still having problems after updating. Hard to know for sure because I was never able to get a consistent repro
